Script file added to configure the test cases

5380harshjain opened the pull request:
Here we are getting the database type and username from the pipeline as parameters,  and we are just validating the database type and running the script. Please review and let us know.

5380harshjain committed:
Script file added to configure the test cases

robe070 commented on Scripts/Ticket_163723.ps1:
```diff
@@ -0,0 +1,45 @@
+﻿
+param (
+    [string] $dbType,
```
This needs to be a list.

5380harshjain replied:
Done

robe070 commented on Scripts/Ticket_163723.ps1:
```diff
@@ -0,0 +1,45 @@
+﻿
+param (
+    [string] $dbType,
+    [string] $username 
```
Why do you need username?

robe070 replied:
@5380harshjain said:

> My understanding here is that we've to map the DB types with LU names in the script and then remove semicolons according to database type and lu name. So can we create key-value {dbType: LU_Name} pair in the script which will contain database type as key and lu_name as value and then iterate over all the database types which will be coming from pipeline and then check it with key-value pair and removing semicolons accordingly from the respective line. Is it the right approach? Please let me know.

Yes that sounds like the correct approach.

And I want to see Log Messages!

5380harshjain replied:
@robe070 I've made some changes, please review and let me know.

robe070 commented on Scripts/Ticket_163723.ps1:
```diff
@@ -0,0 +1,45 @@
+
+
+if($file_data -eq $databaseType){
+    echo("Database Type is correct")
```
Not echo. Use Write-Host. See post in Teams that I wrote on this. Very important!

5380harshjain replied:
Done.

robe070 commented on Scripts/Ticket_163723.ps1:
```diff
@@ -0,0 +1,45 @@
+<#
+foreach($line in [System.IO.File]::ReadLines("C:\Program Files (x86)\Lansa\Verifier_Connection.dat"))
+{
+    Write-Output $line
```
Write-Host! Otherwise you'll get messages coming out in the wrong order.

5380harshjain replied:
Done

robe070 commented on Scripts/Ticket_163723.ps1:
```diff
@@ -0,0 +1,45 @@
+    echo("Database Type is correct")
+    &  "$DbTestScript" -compile $true -test $true
+}
+
```
You are not going to find a match of database type with the lines in verifier_Connection.dat. There is a mapping. See the wiki or the Issue. Its somewhere.

robe070 replied:
The documentation on verifier_connection.dat is here:
https://github.com/lansa/db-regression/wiki#pipeline:~:text=The%20MSSQL%20configuration,execute%20the%20tests

5380harshjain replied:
Done.

5380harshjain committed:
Mapped DB types with LU Names

robe070 commented on Scripts/Ticket_163723.ps1:
```diff
@@ -1,45 +1,52 @@
 param (
-    [string] $dbType,
-    [string] $username 
+    [string[]] $dbTypes
```
mandatory parameter

5380harshjain replied:
Done.

robe070 commented on Scripts/Ticket_163723.ps1:
```diff
@@ -1,45 +1,52 @@
+}
 
+# DB test script 
+$DbTestScript = 'C:\Program Files (x86)\AZURESQL\LANSA\VersionControl\Scripts\db-test.ps1'
```
This is only a true test if 'C:\Program Files (x86)\LANSA is the system being used. That's where Verifier_Connection.dat is.

And I presume you understand this is just a test. Running of the test should not happen here.

5380harshjain replied:
Done.

robe070 commented on Scripts/Ticket_163723.ps1:
```diff
@@ -1,45 +1,52 @@
+        foreach ($string in (Get-Content 'C:\Program Files (x86)\Lansa\Verifier_Connection.dat'))
+        {
+            if($string.Contains($DB_LU_Map[$db])){
+                Write-Host("Changes made in $db")
```
What changes? How does the reader of the log message understand what's been done?

You need a log message before here that says 'Configuring <full path to dat file>'

Then this message needs to be 'Enabling SuperServer test for $db'.

And you need to add a loop which first ensures that all connections HAVE a ';'.

5380harshjain replied:
@robe070  I've made the mentioned changes, please review and let me know if you need anything else to be changed or anything I can improve in the script.

5380harshjain committed:
Feedback Incorporated

robe070 commented on Scripts/configure-verifier-connection.ps1:
```diff
@@ -0,0 +1,52 @@
+# dbTypes is containing database type names which will be coming from pipeline
+param (
+    [parameter(Mandatory=$true)]
+    [string[]] $dbTypes
```
still not Mandatory

robe070 replied:
I'm blind! I'm used to it on the same line. Its fine as it is.

5380harshjain replied:
oh, no issues. Thanks @robe070

robe070 commented on Scripts/configure-verifier-connection.ps1:
```diff
@@ -0,0 +1,52 @@
+}
+
+
+Write-Host("Configuring... $VerifierConnectionPath")
```
Just make the comment a log message. Replace all these lines:
```
Write-Host("Configuring... $VerifierConnectionPath")

#Iterarting over db types to remove the semicolon from particular line from verifier_connection.dat file
```
with
```
Write-Host("Iterating over db types to remove the semicolon from matching lines in $VerifierConnectionPath")
```

Thats what I said to do in the meeting.

5380harshjain replied:
Done.

robe070 commented on Scripts/configure-verifier-connection.ps1:
```diff
@@ -0,0 +1,52 @@
+        foreach ($string in (Get-Content $VerifierConnectionPath))
+        {
+            if($string.Contains($DB_LU_Map[$db])){
+                Write-Host("Enabling SuperServer test for $db")
```
Combine these. USE THE LOG MESSAGES AS COMMENTS! Thats what I said.
```
                Write-Host("Enabling SuperServer test for $db by removing the semicolon")
```

5380harshjain replied:
Done.

robe070 commented:
>  I've made the mentioned comments as log messages, but do we have to write any comments? I was writing it just for reference, currently I've removed all the extra comments.

Almost anything that might be a comment should be a Write-Host.
That is write log messages instead of comments.
Why double up?
Its seems clear to me. I don't understand your confusion.

5380harshjain commented:
> > I've made the mentioned comments as log messages, but do we have to write any comments? I was writing it just for reference, currently I've removed all the extra comments.
> 
> Almost anything that might be a comment should be a Write-Host. That is write log messages instead of comments. Why double up? Its seems clear to me. I don't understand your confusion.

I was writing the comments for developer reference for the future, but I've changed the mentioned comments into logs and removed the others not required.

5380harshjain committed:
Comments converted to logs
